frontend/appflowy_flutter/lib/workspace/presentation/home/panes/flowy_pane_group.dart

```diff
@@ -0,0 +1,143 @@
+class FlowyPaneGroup extends StatelessWidget {
+  final PaneNode node;
+  final double groupWidth;
+  final double groupHeight;
+  final HomeLayout layout;
+  final HomeStackDelegate delegate;
+  const FlowyPaneGroup({
+    super.key,
+    required this.node,
+    required this.groupWidth,
+    required this.groupHeight,
+    required this.layout,
+    required this.delegate,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    if (node.children.isEmpty) {
+      return Listener(
+        behavior: HitTestBehavior.translucent,
+        onPointerDown: (_) => context.read<PanesCubit>().setActivePane(node),
+        child: FlowyPane(
+          node: node,
+          delegate: delegate,
+          layout: layout,
+          paneContext: context,
+          size: Size(groupWidth, groupHeight),
+        ),
+      );
+    }
+
+    return BlocProvider(
+      key: ValueKey(node.paneId),
+      create: (context) => PaneNodeCubit(node.children.length),
+      child: BlocBuilder<PaneNodeCubit, PaneNodeState>(
+        builder: (context, state) {
+          return SizedBox(
+            child: LayoutBuilder(
+              builder: (context, constraints) {
+                return Stack(
+                  children: [
+                    ...node.children.indexed.map((indexNode) {
+                      final paneLayout = PaneLayout(
+                        childPane: indexNode,
+                        parentPane: node,
+                        flex: state.flex,
+                        parentPaneConstraints: constraints,
+                      );
+                      return Stack(
+                        children: [
+                          _resolveFlowyPanes(paneLayout, indexNode),
+                          _resizeBar(
+                            indexNode,
+                            context,
+                            paneLayout,
+                          )
+                        ],
+                      );
+                    }).toList(),
+                  ],
+                );
+              },
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  Widget _resizeBar(
+    (int, PaneNode) indexNode,
+    BuildContext context,
+    PaneLayout paneLayout,
+  ) {
+    return Positioned(
+      left: paneLayout.childPaneLPosition,
+      top: paneLayout.childPaneTPosition,
+      child: MouseRegion(
+        cursor: paneLayout.resizeCursorType,
+        child: GestureDetector(
+          dragStartBehavior: DragStartBehavior.down,
+          onHorizontalDragUpdate: (details) =>
+              context.read<PaneNodeCubit>().resize(
+                    paneLayout.childPaneWidth,
+                    indexNode.$1,
+                    details.delta.dx,
+                  ),
+          onVerticalDragUpdate: (details) =>
+              context.read<PaneNodeCubit>().resize(
+                    paneLayout.childPaneHeight,
+                    indexNode.$1,
+                    details.delta.dy,
+                  ),
```

Comment:
I'm not sure what's a good way to improve this, I'll need to look at it in-depth.

Could we store the width of one of the panes at drag start, and then get the difference on drag to only change the width of the pane like that, or is this not viable?

Currently the drag behavior feels quite out of place since it doesn't match the cursor.

Reply:
Yeah that would be a bit difficult because right now there is a List<double> in all the branch nodes, which equates flex for each consecutive child node, since having a double value for width in each leaf node(that is being rendered) meant I would have to do a whole tree traversal to recalculate all the other widths which would have been more expensive.

So storing initial offset for each pane would mean usage of another array and then modifying that, major struggle of this entire problem is finding the right way to convert the offset change to change in width / flex

I can give it a shot though to see if that might bring improvements.

Reply:
Hey Mathias I tried this out, approach didn't do much for optimization however I was able to define a better algorithm for flex adjustment, from all the things I could try out I couldn't optimize it anymore it still does feel heavy in debug builds, there are a lot of junk frames however practically it feels pretty smooth when I was inspecting expensive parts in profile mode.

What do you think, here is a vid



https://github.com/AppFlowy-IO/AppFlowy/assets/84044317/15bb3596-5aca-48e4-a366-80d8ccf668e2

Reply:
This looks really good compared to the old one, I'll try it out :pray: